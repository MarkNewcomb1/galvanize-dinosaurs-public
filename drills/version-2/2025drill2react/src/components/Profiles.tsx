import { useState } from "react";
import { Dino } from '../types/types'

  type ProfilesProps = {
    dinos: Dino[];
  };
  
  const Profiles = ({dinos}: ProfilesProps) => {
    // Record means an object where the keys are numbers and the values are booleans.
    const [showSkills, setShowSkills] = useState<Record<number, boolean>>({});

    const toggleSkills = (index: number) => {
        setShowSkills((prev) => ({
          ...prev, // copy everything
          // using value of index as the key in the object. 
          // Flip the current value at key passed in as index and overwrite it in the object
          [index]: !prev[index],
        }));
      };
    return (
      <>
      {dinos.map((dino, index) => (
        <div key={index} 
        className="profile-card" 
        onClick={() => toggleSkills(index)}
        style={{ cursor: "pointer" }}
        >
        <header className="profile-header">
          <img src={dino.image} />
          <h2>{dino.name}</h2>
        </header>
        <section className={
              showSkills[index]
                ? "skills-container"
                : "skills-container hidden"
            }>
          <h4>Skills</h4>
          <ul className="skills-list">
            {dino.skills.map((skill, index) => (
                <li key={index}>{skill}</li>
            ))}
          </ul>
        </section>
      </div>
      ))}
      </>
    )
  }
  
  export default Profiles