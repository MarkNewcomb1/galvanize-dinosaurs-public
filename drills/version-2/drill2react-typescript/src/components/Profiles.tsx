import React from 'react';
import { Dinosaur } from '../App';
import SkillList from './SkillList';

interface Props {
    showSkills: boolean | string;
    profiles: Array<Dinosaur>;
    toggleVisibility: (name: string) => void;
}

const Profiles: React.FC<Props> = ({ showSkills, profiles, toggleVisibility }): any => {
    return (
        profiles.map(({ name, skills, image }) => {
            return (
                <li key={name} onClick={() => toggleVisibility(name)}>
                    <div className="profile-card">
                        <header className="profile-header">
                            <img src={image} alt="dino" />
                            <h2>{name}</h2>
                        </header>
                        <section className={showSkills == name ? "skills-container" : "skills-container hidden"}>
                            <h4>Skills</h4>
                            <ul className="skills-list">
                                <SkillList skills={skills} />
                            </ul>
                        </section>
                    </div>
                </li>
            )
        })
    )
}

export default Profiles;