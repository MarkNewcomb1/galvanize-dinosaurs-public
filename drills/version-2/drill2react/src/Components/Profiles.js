import React from 'react';
import SkillList from './SkillList';

const Profiles = ({showSkills, profiles, toggleVisibility}) => {
    return (
        profiles.map(({name, skills, image}) => {
            return (
                <li key={name} onClick={ () =>toggleVisibility(name)}>
                    <div className="profile-card">
                        <header className="profile-header">
                            <img src={image} alt="dino" />
                            <h2>{name}</h2>
                        </header>
                        <section className={ showSkills == name ? "skills-container" : "skills-container hidden" }>
                            <h4>Skills</h4>
                            <ul className="skills-list">
                                <SkillList skills={skills} />
                            </ul>
                        </section>
                    </div>
                </li>
            )
        })
    )
}

export default Profiles;