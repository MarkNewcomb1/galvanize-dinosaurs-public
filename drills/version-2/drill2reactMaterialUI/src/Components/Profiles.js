import React from 'react';
import SkillList from './SkillList';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const Profiles = ({showSkills, profiles, toggleVisibility}) => {
    return (
        profiles.map(({name, skills, image}) => {
            return (
                <li key={name} onClick={ () =>toggleVisibility(name)} style={{marginBottom: '10px'}}>
                    <Paper 
                    elevation={12}
                    square={true}
                    className="profile-card">
                        <header className="profile-header">
                            <img src={image} alt="dino" />
                            <Typography
                            variant="title"
                            >{name}</Typography>
                        </header>
                        <section className={ showSkills === name ? "skills-container" : "skills-container hidden" }>
                            <Typography
                            variant="subheading"
                            >Skills</Typography>
                            <ul className="skills-list">
                                <SkillList skills={skills} />
                            </ul>
                        </section>
                    </Paper>
                </li>
            )
        })
    )
}

export default Profiles;