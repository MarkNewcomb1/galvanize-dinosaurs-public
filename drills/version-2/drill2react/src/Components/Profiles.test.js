import React from 'react';
import Profiles from "./Profiles";
import Enzyme, {render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('There are three profiles', () => {
    const component = render(
        <Profiles
        profiles={[{"name": "T-Rex", "image": "./assets/dino-1.png", "skills": ["stepping on things", "musing on life", "MS Office"]},{"name": "Utahraptor", "image": "./assets/dino3.png", "skills": ["being contrary"]},{"name": "Dromiceiomimus", "image": "./assets/dino-2.png", "skills": ["Java", "Ninja", "MongoDB", "Git and Github", "Agile"]}]
        }
        showSkills={false}
        toggleVisibility={"T-Rex"}
        />
    );

    expect(component).toHaveLength(3);
});
