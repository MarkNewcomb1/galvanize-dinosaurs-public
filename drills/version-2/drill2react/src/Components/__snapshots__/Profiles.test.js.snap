// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profiles have skills hidden by default 1`] = `
Array [
  <li
    onClick={[Function]}
  >
    <div
      className="profile-card"
    >
      <header
        className="profile-header"
      >
        <img
          alt="dino"
          src="./assets/dino-1.png"
        />
        <h2>
          T-Rex
        </h2>
      </header>
      <section
        className="skills-container hidden"
      >
        <h4>
          Skills
        </h4>
        <ul
          className="skills-list"
        >
          <li>
            stepping on things
          </li>
          <li>
            musing on life
          </li>
          <li>
            MS Office
          </li>
        </ul>
      </section>
    </div>
  </li>,
  <li
    onClick={[Function]}
  >
    <div
      className="profile-card"
    >
      <header
        className="profile-header"
      >
        <img
          alt="dino"
          src="./assets/dino3.png"
        />
        <h2>
          Utahraptor
        </h2>
      </header>
      <section
        className="skills-container hidden"
      >
        <h4>
          Skills
        </h4>
        <ul
          className="skills-list"
        >
          <li>
            being contrary
          </li>
        </ul>
      </section>
    </div>
  </li>,
  <li
    onClick={[Function]}
  >
    <div
      className="profile-card"
    >
      <header
        className="profile-header"
      >
        <img
          alt="dino"
          src="./assets/dino-2.png"
        />
        <h2>
          Dromiceiomimus
        </h2>
      </header>
      <section
        className="skills-container hidden"
      >
        <h4>
          Skills
        </h4>
        <ul
          className="skills-list"
        >
          <li>
            Java
          </li>
          <li>
            Ninja
          </li>
          <li>
            MongoDB
          </li>
          <li>
            Git and Github
          </li>
          <li>
            Agile
          </li>
        </ul>
      </section>
    </div>
  </li>,
]
`;
